SuperdenseCoding/Tasks.qs

```diff
@@ -38,24 +43,22 @@ namespace Quantum.Kata.SuperdenseCoding {
     // Finally, we compose those steps into the complete superdense coding protocol.
     
     // Task 1. Entangled pair
-    // Input: An array of two qubits in the |00⟩ state.
-    // Goal:  Create a Bell state |Φ⁺⟩ = (|00⟩ + |11⟩) / sqrt(2) on these qubits.
-    operation CreateEntangledPair (qs : Qubit[]) : Unit {
-        // The following lines enforce the constraints on the input that you are given.
-        // You don't need to modify them. Feel free to remove them, this won't cause your code to fail.
-        EqualityFactI(Length(qs), 2, "The array should have exactly 2 qubits.");
+    // Input: A tuple of two qubits.
+    // Goal:  If both qubits are initially in a state |00>, 
```

Comment:
This task description is incorrect/incomplete; we don't ask people to figure out whether the qubits are in state 00 and modify the state only if they are, we ask them to prepare a certain state and guarantee that the qubits start in 00.

If you want to modify this task to ask for a unitary transformation H-CNOT, you need to properly describe it, and to modify the test `T1_CreateEntangledPair_Test` to use `AssertOperationsEqualReferenced` instead of `AssertEqualOnZeroState` to make sure that it checks exactly the same thing as the task asks to do.

If you want to keep this task as is (which, to be fair, is what I would do - we don't *force* people to use adjoint of this task to perform later stages of the teleportation protocol, and formulating the task as state preparation doesn't prevent *us* from doing it our way and using it later), keep the goal as it was and modify input to "Two qubits, each in the |0⟩ state."

(Also, note the use of fancy Unicode |0⟩ instead of plain ASCII |0> to denote states :-) )

Reply:
Fair point. As suggested, I kept the original task goal and modified the input description to "Two qubits, each in the |0⟩ state.".